pkg/policies/dataplane/resources/classifier/provide.go

```diff
@@ -109,24 +115,41 @@ func invokeMiniApp(
 	unmarshaller config.Unmarshaller,
 	classifier *Classifier,
 ) error {
-	var rs classificationv1.Classifier
-	if err := unmarshaller.Unmarshal(&rs); err != nil {
+	wrapperMessage := &wrappersv1.ClassifierWrapper{}
+	err := unmarshaller.Unmarshal(wrapperMessage)
+	if err != nil || wrapperMessage.Classifier == nil {
+		log.Warn().Err(err).Msg("Failed to unmarshal classifier config wrapper")
 		return err
 	}
-
 	var activeRuleset ActiveRuleset
-
+	rs := wrapperMessage.Classifier
+	classifier.classifierProto = rs
+	classifier.policyName = wrapperMessage.PolicyName
+	classifier.policyHash = wrapperMessage.PolicyHash
+	classifier.classifierIndex = wrapperMessage.ClassifierIndex
 	lc.Append(
 		fx.Hook{
 			OnStart: func(startCtx context.Context) error {
 				var err error
-				activeRuleset, err = classifier.AddRules(startCtx, string(key), &rs)
+				activeRuleset, err = classifier.AddRules(startCtx, string(key), rs)
 				if err != nil {
 					return err
 				}
+
+				// Register metric with PCA
+				err = engineAPI.RegisterClassifier(classifier)
```

Comment:
@DariaKunoichi,

This implementation is incorrect. Classifier runs in the Authz stage. And it may manipulate the labels. By the time engine runs, the labels are already mutated so it might not give the same results. On top of that we are unnecessarily running multi matcher for Classifiers twice.

The matched Classifiers should be encoded in `flowcontrol.AuthzResult` (I'll be renaming it from `AuthzResponse`)

cc: @jmichalak-fluxninja